### PR TITLE
GN-176 Optional padding on back to top

### PR DESCRIPTION
**Pull request by johndavey72**
https://nicedigital.atlassian.net/browse/GN-176

### CODE REVIEW

File: src/Main/Main.jsx [hunk 1]
```diff
@@ -6,10 +6,17 @@ import classnames from "classnames";
 import styles from "./Main.module.scss";
 
 export function Main(props) {
-	const { children, className, ...rest } = props;
+	const { children, className, withPadding, ...rest } = props;
```
Comment (ediblecode) [inline src/Main/Main.jsx:9]: Should `withPadding` default to `true`? I think the most common scenario is that we _do_ want padding, so it'd be 'opt out' if you had wanted to remove it?

```suggestion
	const { children, className, withPadding = false, ...rest } = props;
```

File: src/Main/Main.module.scss [hunk 2]
```diff
@@ -4,9 +4,15 @@ $back-to-top-height: rem(
...
 
+$main-content-padding: 4rem;
+
 @supports (position: sticky) {
   .main {
     padding-bottom: $back-to-top-height; /* space for the back to top link to dock */
     position: relative;
+
+    &.withPadding {
+      padding-bottom: $main-content-padding; /* space between page content and footer when withPadding prop set to true */
```
Comment (ediblecode) [inline src/Main/Main.module.scss:15]: I wonder if this makes more sense to be `$back-to-top-height + rem($nds-spacing-large)`? As in the the `$main-content-padding` variable isn't clear if it's the padding on the main element or the gap above the back to top button, so doing the addition here makes this clearer

File: src/Main/BackToTop/BackToTop.module.scss [hunk 3]
```diff
@@ -31,6 +31,7 @@ $back-to-top-height: rem(
   }
 
   svg {
+    margin-right: 0.5rem;
```
Comment (ediblecode) [inline src/Main/BackToTop/BackToTop.module.scss:34]: can we make this a `rem($nds-spacing-small)`?

File: src/Main/BackToTop/BackToTop.module.scss [hunk 4]
```diff
@@ -41,6 +42,10 @@ $back-to-top-height: rem(
       transform: translateY(-#{rem($nds-spacing-xx-small)});
     }
   }
+
+  .container {
+    display: flex;
```
Comment (ediblecode) [inline src/Main/BackToTop/BackToTop.module.scss:47]: I think we should probably have `align-items: center;` here too to makes sure the svg icon always aligns with the middle of the text

File: README.md [hunk 5]
```diff
@@ -330,6 +332,15 @@ For a full list of all the available props, see the [props section](#props) belo
 
 #### Props
 
+##### Main props
+
+###### Main.withPadding
+
+- Type: `Boolean`
+- Default: false
```
Comment (ediblecode) [inline README.md:340]: I think the default would/should be `true` as that's the more common scenario.

**Review by ediblecode — CHANGES_REQUESTED** [review]

File: src/Main/Main.jsx (hunk 1 above)
Comment (johndavey72) [reply src/Main/Main.jsx]: can you double check I haven't misunderstood the comment above please Ian? I've defaulted to true on line 9...

**Review by ediblecode — APPROVED** [review]

**Pull request merged**